### ISSUE TITLE
Author affiliations

### ISSUE THREAD

**Issue by fred-atherden** [libero/editor#28]
Structured affiliation information (department, institution, city, country).

### Author
- [x] As an author, I want to be able to clearly see which affiliations are associated with an author so that I can check this information is correct.
- [x] As an author, I want to be able to see which authors are associated with an affiliation so that I can check they have been correctly assigned. 
- [x] As an author, I want to be able to add or remove affiliations from an author so that I can correct any errors in affiliation attribution.
- [x] As an author, I want to be able to add or remove authors from an affiliation so that I can correct any errors in affiliation attribution.
- [x] As an author, I want to be able to edit an affiliation so that I can correct any errors.
- [x] As an author, I want to be able to create a new affiliation and associate it with existing authors so that I can correct affiliation information.
- [x] As an author I want to be able to delete affiliations so that I can remove any added in error.

### Production staff
- [x] As production staff, I want to be able to clearly see the divisions between different parts of an affiliation so that I can check it has been structured correctly.

### But what if . . . ?
- 

### Considerations
- We will have a list of authors and a list of multiple affiliations (in most cases) with a many-to-many relationship between them.
- Different journals divide affiliations up in different ways. eLife uses Department, Institution, City, and Country. Érudit will be using the more generic Organisation division 1, Organisation division 2, Organisation division 3 . . ., Organisation name, City, Country.
- The consortium has agreed to place affiliations within the contrib-group containing the authors, rather than outside it, and to using xref linking to connect authors to affiliations. This tagging will be use for non-author contrib-groups as well. E.g. for every contrib-group, there will be a unique set of affiliations.
- For eLife, we order the affiliations based on the order in which the authors appear e.g. the first author's affiliation will be aff1, the second authors' affiliation will be aff2 etc.
- Given the requirement that each ID be unique, affiliation IDs in other contrib-groups should follow on from the IDs in the author/first contrib-group.
- eLife adds labels to affiliations for PDF and PMC display. Érudit and other publishers do not do this.

## XML requirements
```xml
<contrib-group content-type="author">
       <contrib contrib-type="author">
              <name>
                     <surname>Surname</surname>
                     <given-names>First-name Middle name</given-names>
              </name>
              <xref ref-type="aff" rid="aff1">1</xref>
              <xref ref-type="aff" rid="aff2">2</xref>
              ...
       </contrib>
       <aff id="aff1">
              <label>1</label>
              <institution>Department of Production, eLife</institution>
              <city>Cambridge</city>
              <state>Cambridgeshire</state>
              <country country="GB">United Kingdom</country>
       </aff>
       <aff id="aff2">
              <label>2</label>
              <institution>Department of Technology, eLife</institution>
              <city>Cambridge</city>
              <country country="GB">United Kingdom</country>
       </aff>       
</contrib-group>      
```

The value of the country attribute should be one of the [ISO_3166-1](https://en.wikipedia.org/wiki/ISO_3166-1) 2 letter country codes, although the text inside the country element does not need to follow the respective English name in that list.

Editor also needs to account for affiliations that can be broken down using `orgdivX` and `orgname` values for `@content-type`. For example:
```xml
<aff id="aff1">
      <institution content-type="orgname">German Primate Center GmbH</institution>
      <institution content-type="orgdiv1">Neurobiology Laboratory</institution>
      <city>Göttingen</city>
      <country>Germany</country>
</aff>
```

Taken from: https://github.com/elifesciences/TextureRequirements/issues/14

**Comment by JGilbert-eLife** [comment]
Something we didn't talk about - states. There is a tag for this `<state>` but I'm not sure if we want to use it. We occasionally get authors complain we don't include this information, but we decided from the get go not to do so.

Something to revisit?

**Comment by Melissa37** [comment]
>Something we didn't talk about - states. There is a tag for this <state> but I'm not sure if we want to use it. We occasionally get authors complain we don't include this information, but we decided from the get go not to do so.

Good point, I think one of our deputy editors is very keen to introduce this. My problem is that it would be US-focussed only.
We could sound out the data and editorial teams?

**Comment by JGilbert-eLife** [comment]
Not just US - affs from China and Japan could also include this.

And yes, I think sounding out those teams is a good idea.

**Comment by NickDuf** [comment]
- This feature is not available in Texture, but we assume it to follow the same pattern as affiliations which is
- There is some desire to make affiliation a two-way editing and creation feature which is not currently supported in texture
- Texture currently supports more inputs than are needed by eLife. [See inVision](https://projects.invisionapp.com/share/MCSFHRIN9R4#/screens/367881081_affiliations-Desktop) file
